### ISSUE TITLE
Weird behavior of mod after cloning the constructor

### ISSUE THREAD

**Issue by josdejong** [MikeMcl/decimal.js#26]
I encountered some weird behavior in decimal.js v5:

```js
console.log('65 % 40 = ' + new Decimal(65).mod(40));   // 25, correct

var D = Decimal.clone({});  
console.log('65 % 40 = ' + new D(65).mod(40));         // -15, not correct
```

any idea?

**Comment by MikeMcl** [comment]
Oh, I can see that I haven't considered that some or all of the config properties may be omitted when passing an object to `clone`.

In your case, it looks like the modulo mode is therefore being interpreted as zero.

I will fix ASAP. In the meantime, `clone` will work fine if called without any argument, optionally followed by a call to `config`.

Thanks for the report. I was about to watch a film. No longer it seems.

**Comment by josdejong** [comment]
Ok great, thanks for your fast response.

The workaround is indeed easy.

**Comment by MikeMcl** [comment]
Fixed in v5.4.0. 

Cheers, Jos.

**Comment by josdejong** [comment]
Awesome, thanks! (typo, it's v5.0.4)

I'm really happy with the new trigonometric functions in v5, the implementations we have in math.js needed a serious refactoring but now we can simply replace them with the implementations of decimal.js :D.

**Comment by josdejong** [comment]
@MikeMcl Turns out there is still an issue here. I couldn't exactly pinpoint what it is but you can reproduce it like this:

```js
// Wolfram: acos(1/3)= 1.230959417340774682134929178247987375710340009355094839055548334


Decimal.config({precision: 30});
console.log(new Decimal(1).div(3).acos().toString()); // 1.23095941734077468213492917825
                                                      // CORRECT
Decimal.config({precision: 20});

var D = Decimal.clone({precision: 30});
console.log(Decimal.precision); // 20
console.log(D.precision);       // 30
console.log(new D(1).div(3).acos().toString());       // 1.23095941734077468214045511458
                                                      // WRONG

// output of D is changed after changing the precision of Decimal
Decimal.config({precision: 30});
console.log(Decimal.precision); // 30
console.log(D.precision);       // 30
console.log(new D(1).div(3).acos().toString());       // 1.23095941734077468213492917825
                                                      // CORRECT
```

**Comment by josdejong** [comment]
Thanks again for your fast response!

**Comment by josdejong** [comment]
just wondering, what is the value of using `new Ctor(ONE)` over `new Ctor(1)`?

**Comment by MikeMcl** [comment]
That would mean storing it as a property on each clone, i.e. restoring the `Decimal.ONE` property, which I removed because I thought it just unnecessary bloat in the API. And then it would follow that each clone would have to have their own copy of `LN10` and `PI` which would be a bit much...

I've changed my mind since my earlier comment, I'm going to keep it internally, but, yes, it is certainly something that needs to handled carefully as the number of tests involving multiple constructors are limited at present.
